src/cmd/serve.rs

```diff
@@ -102,6 +102,24 @@ fn create_new_site(interface: &str, port: &str, output_dir: &str, base_url: &str
     Ok((site, address))
 }
 
+/// Attempt to render `index.html` when a directory is requested.
+///
+/// The default "batteries included" mechanisms for actix to handle directory
+/// listings rely on redirection which behaves oddly (the location headers
+/// seem to use relative paths for some reason).
+/// They also mean that the address in the browser will include the
+/// `index.html` on a successful redirect (rare), which is unsightly.
+///
+/// Rather than deal with all of that, we can hijack a hook for presenting a
+/// custom directory listing response and serve it up using their
+/// `NamedFile` responder.
+fn handle_directory<'a, 'b>(dir: &'a fs::Directory, req: &'b HttpRequest) -> io::Result<HttpResponse> {
+    let mut path = PathBuf::from(&dir.base);
+    path.push(&dir.path);
+    path.push("index.html");
+    Ok(fs::NamedFile::open(path).respond_to(req).unwrap())
```

Comment:
Looking back on the diff, this largely doesn't seem correct. `unwrap`ping then rewrapping should be harmful _and_ redundant. Looks like the error type that comes from `respond_to` isn't an `io::Error` either -- let me map it real quick!

Reply:
@fafhrd91 will do. I'll write something up soon.

Reply:
@fafhrd91 would it make sense to have actix-web be stand alone and re-export the bits it needs from actix so you only have 1 package to add to get started?